addons/purchase_stock/tests/test_purchase_order_suggest.py

```diff
@@ -319,6 +320,41 @@ def test_purchase_order_suggest_quantities(self):
             {'product_id': product_4.id, 'product_qty': 2},
         ])
 
+        # ------ Check pricelist selection ------
+        product_7 = self.env['product.product'].create({
+            'name': 'Product 7',
+            'standard_price': 12,
+            'is_storable': True,
+        })
+        self.env['product.supplierinfo'].create({
+            'partner_id': self.partner_1.id,
+            'price': 17,
+            'product_id': product_7.id,
+            'min_qty': 2
+        })
+        self.env['product.supplierinfo'].create({
+            'partner_id': self.partner_1.id,
+            'price': 13,
+            'product_id': product_7.id,
+            'min_qty': 3
+        })
+        self.env['stock.quant']._update_available_quantity(product_7, self.stock_location, 1)
```

Comment:
Move that part at the beginning of the test and use multi-create for the record creation